sound/soc/sof/intel/hda.c

```diff
@@ -597,6 +600,9 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 	struct pci_dev *pci = to_pci_dev(sdev->dev);
 	const struct sof_intel_dsp_desc *chip = hda->desc;
 
+	/* First Cancel any pending attempt to put DSP to D0i3 */
+	cancel_delayed_work_sync(&hda->d0i3_work);
```

Comment:
don't you also need to cancel this work during system suspend?

Reply:
you mean D0I3 suspend? that is already covered by hda_dsp_suspend()->hda_dsp_set_power_state() here:
```
	case SOF_DSP_STATE_D0I3:
		/* First Cancel any pending attempt to put DSP to D0i3 */
		cancel_delayed_work_sync(&hda->d0i3_work);

		/* disable DMA trace for no delay D0I3 */
		ret = hda_dsp_set_d0i3(sdev, HDA_PM_NO_DMA_TRACE);

```